packages/astro/components/ViewTransitions.astro

```diff
@@ -128,7 +128,12 @@ const { fallback = 'animate' } = Astro.props as Props;
 			}
 		});
 		window.addEventListener('popstate', () => {
-			if (!transitionEnabledOnThisPage()) return;
+			if (!transitionEnabledOnThisPage()) {
+				// The current page doesn't haven't View Transitions,
+				// respect that with a full page reload
+				location.reload();
```

Comment:
I was unsure if this was the best way but from research it seems like it probably is? popstate is a weird API.

Reply:
I suppose that's fine